app/models/katello/upstream_pool.rb

```diff
@@ -4,9 +4,15 @@ class UpstreamPool < OpenStruct
 
     class << self
       def fetch_pools(params)
-        response = CP_POOL.get(params: cp_request_params.deep_merge(params))
+        pool_ids = params["pool_ids"] ? pool_id_params(params["pool_ids"]) : []
+        params.delete("pool_ids")
 
+        cp_params = request_params(base_params: params,
+                                   extra_params: pool_ids,
```

Comment:
how about:

```
extra_params: pool_id_params(params.delete(:pool_ids))
```
and have pool_id_params return [] if it receives nil?

reads a bit cleaner, removes the ternary